gst/nnstreamer/nnstreamer_plugin_api_util_impl.c

```diff
@@ -511,6 +511,9 @@ gst_tensors_info_copy (GstTensorsInfo * dest, const GstTensorsInfo * src)
   num = dest->num_tensors = src->num_tensors;
   dest->format = src->format;
 
+  if (src->format != _NNS_TENSOR_FORMAT_STATIC)
+    return;
+
```

Comment:
If we look at the API description, it mentions using gst_tensors_info_free() after using this API.

However, if the format is not ```_NNS_TENSOR_FORMAT_STATIC```,  ```gst_tensor_info_copy()``` is not performed, so it seems that ```gst_tensors_info_free()``` should not be performed.
It seems that the comment needs to be edited or code added to check the format in ```gst_tensors_info_free()```.
Right?

```
/**
 * @brief Copy tensor info
 * @note Copied info should be freed with gst_tensors_info_free()
 */
void
gst_tensors_info_copy (GstTensorsInfo * dest, const GstTensorsInfo * src)
{
[...]
 + if (src->format != _NNS_TENSOR_FORMAT_STATIC)
 +   return;
    gst_tensor_info_copy (_dest, _src);
}
```

Reply:
I think `gst_tensors_info_init` is called in L511, and makes the free function good to go.

Reply:
Thank you, I got it. 
gst_tensors_info_init (info) is called twice, but it is not a problem.
```
/**
 * @brief Free allocated data in tensors info structure
 * @param info tensors info structure
 */
void
gst_tensors_info_free (GstTensorsInfo * info)
{
  guint i;

  g_return_if_fail (info != NULL);

  for (i = 0; i < NNS_TENSOR_MEMORY_MAX; i++) {
    gst_tensor_info_free (&info->info[i]);
  }

  if (info->extra) {
    for (i = 0; i < NNS_TENSOR_SIZE_EXTRA_LIMIT; ++i)
      gst_tensor_info_free (&info->extra[i]);

    g_free (info->extra);
    info->extra = NULL;
  }

  /* Init default */
  gst_tensors_info_init (info);
}
```

Reply:
Let's refactor it later. :smile: